File: TP1/palindromorec.java
class palindromorec{
    public static boolean isFim(String s){
        return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    public static boolean palindromo(String s){
        return palindromoRec(s , 0 , (s.length()-1));
    }   

    public static boolean palindromoRec(String s , int i , int j){
        boolean resp = true;
       
        if(j >= i){
            if(s.charAt(i) != s.charAt(j)){
                resp = false;
            }else{
                resp = palindromoRec(s, i+1, j-1);             
            }
        }

        return resp;
    }

    public static void main(String[] args) {

        String[] entrada = new String[1000];

        do{
            entrada[0] = MyIO.readLine();
            if(palindromo(entrada[0])){
                System.out.println("SIM");
            }else{
                System.out.println("NAO");
            }
        } while (isFim(entrada[0]) == false);
 
    }
}

File: TP1/PalindromoRec.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

#define ENTRADA 1000
#define TAM   1000

bool isFim(char *string){
    return (strlen(string) >= 3 && string[0] == 'F' && string[1] == 'I' && string[2] == 'M');
}

bool isPalindromo(char* original, int i, int j){
    bool resp = true;

    if(j >= i){
        if(original[i] != original[j]){
            resp = false;
        }
        else{
            resp = isPalindromo(original, i+1, j-1);
        }
    }

    return resp;
}

int main(){
    
    char linha[ENTRADA][TAM];
    int numEntrada = 0;

    do{
        fgets(linha[numEntrada], ENTRADA, stdin);
    }while(isFim(linha[numEntrada++]) == false);
    numEntrada--;
   
    for(int i = 0; i < numEntrada; i++ ){
        if(isPalindromo(linha[i], 0, (strlen(linha[i])-2)) == true){
            printf("SIM\n");
        }else{
            printf("NAO\n");
        }
    }

    return 0;
}